src/utils/adapter.ts

```diff
@@ -1,47 +1,62 @@
-import PostRobot from 'post-robot';
-import { AxiosRequestConfig, AxiosResponse } from 'axios';
-import { onError, fetchToAxiosConfig } from './utils';
+import PostRobot from "post-robot";
+import { AxiosRequestConfig, AxiosResponse, isAxiosError } from "axios";
+
+import { onError, fetchToAxiosConfig, serializeAxiosResponse } from "./utils";
+import { RequestInitConfig } from "../types/api.type";
 
 /**
  * Dispatches a request using PostRobot.
  * @param postRobot - The PostRobot instance.
  * @returns A function that takes AxiosRequestConfig and returns a promise.
  */
-export const dispatchAdapter = (postRobot: typeof PostRobot) => (config: AxiosRequestConfig)=> {
+export const dispatchAdapter = (postRobot: typeof PostRobot) => (config: AxiosRequestConfig): Promise<AxiosResponse> => {
   return postRobot
-    .sendToParent("apiAdapter",  config )
-    .then(({ data }) => ({ ...data, config }))
+    .sendToParent("apiAdapter", config)
+    .then((event:unknown) => {
+      const { data } = event as { data: AxiosResponse };
+      if (data.status >= 400) {
+        throw serializeAxiosResponse(data, config);
+      }
+      return serializeAxiosResponse(data, config);
+    })
     .catch(onError);
 };
-
 /**
  * Dispatches an API request using axios and PostRobot.
  * @param url - The URL of the API endpoint.
  * @param options - Optional request options.
  * @returns A promise that resolves to a partial Response object.
  */
-export const dispatchApiRequest = async (url: string, options?: RequestInit): Promise<Response> => {
-  try {
-    const config = fetchToAxiosConfig(url, options);
-    const responseData = await dispatchAdapter(PostRobot)(config) as AxiosResponse; 
-    return new Response(responseData.data,{
-      status: responseData.status,
-      statusText: responseData.statusText,
-      headers: new Headers(responseData.config.headers || {}),
-    });
+export const dispatchApiRequest = async (
+    url: string,
+    options?: RequestInitConfig,
+): Promise<Response> => {
+    try {
+        const config = fetchToAxiosConfig(url, options);
+        const responseData = (await dispatchAdapter(PostRobot)(
+            config
+        )) as AxiosResponse;
+  
+        if (isAxiosError(responseData)) {
+          throw responseData;
+        }
+        const response = new Response(responseData?.data, {
+            status: responseData.status,
+            statusText: responseData.statusText,
+            headers: new Headers(responseData.config.headers || {}),
+        });
+        return response
+        
+    } catch (error: any) {
```

Comment:
Why do we need type here? Check whether the error is instance of AxiosError else handle as generic error.

Reply:
Not able to change error type other then unkown or any for catch

Reply:
```suggestion
    } catch (err) {
       if (err instanceof AxiosError) {
    }
```
I meant something of this sort.

Reply:
made the changes